da.go

```diff
@@ -36,6 +36,10 @@ type DA interface {
 // posted to, for DA layers supporting the functionality.
 type Namespace = []byte
 
+// KeyName is an optional parameter used to set the keyring keyname for
```

Comment:
TBH it's  not "optional"  as you  have  to pass something in Go call.

Reply:
To make it actually optional, we should implement `Option` pattern and accept  them using `...` in functions.